### DISCUSSION TITLE
Compile with GraalVM

### DISCUSSION THREAD

**Discussion by hamedsharifi** [Auties00/WhatsappWeb4j#90] [Ideas]
Hello
I decided to compile my app using GraalVM but I got this error:
```
Fatal error:com.oracle.graal.pointsto.util.AnalysisError$ParsingError: Error encountered while parsing org.glassfish.tyrus.core.ServiceFinder$LazyObjectIterator.next()
Parsing context:
   at org.glassfish.tyrus.core.ServiceFinder$LazyObjectIterator.next(ServiceFinder.java:761)
   at org.glassfish.tyrus.core.ComponentProviderService.create(ComponentProviderService.java:62)
   at org.glassfish.tyrus.core.TyrusServerEndpointConfigurator.<init>(TyrusServerEndpointConfigurator.java:40)
   at com.oracle.svm.core.reflect.ReflectionAccessorHolder.TyrusServerEndpointConfigurator_constructor_a96715562e86824a6870e2b7a8a7fec2b0cc497d(generated:0)

```
It seems the websocket library is uncompatible
any suggestion?

**Comment by Auties00** [comment]
> Hello I decided to compile my app using GraalVM but I got this error:
> 
> ```
> Fatal error:com.oracle.graal.pointsto.util.AnalysisError$ParsingError: Error encountered while parsing org.glassfish.tyrus.core.ServiceFinder$LazyObjectIterator.next()
> Parsing context:
>    at org.glassfish.tyrus.core.ServiceFinder$LazyObjectIterator.next(ServiceFinder.java:761)
>    at org.glassfish.tyrus.core.ComponentProviderService.create(ComponentProviderService.java:62)
>    at org.glassfish.tyrus.core.TyrusServerEndpointConfigurator.<init>(TyrusServerEndpointConfigurator.java:40)
>    at com.oracle.svm.core.reflect.ReflectionAccessorHolder.TyrusServerEndpointConfigurator_constructor_a96715562e86824a6870e2b7a8a7fec2b0cc497d(generated:0)
> ```
> 
> It seems the websocket library is uncompatible any suggestion?

I'm writing the MD version currently and I was thinking to switch to Java's built in implementation of web sockets(jdk.http). When it's ready you'll be able to compile with Graal i think. In the meantime, I don't know honestly

**Comment by Auties00** [comment]
> Hello I decided to compile my app using GraalVM but I got this error:
> 
> ```
> Fatal error:com.oracle.graal.pointsto.util.AnalysisError$ParsingError: Error encountered while parsing org.glassfish.tyrus.core.ServiceFinder$LazyObjectIterator.next()
> Parsing context:
>    at org.glassfish.tyrus.core.ServiceFinder$LazyObjectIterator.next(ServiceFinder.java:761)
>    at org.glassfish.tyrus.core.ComponentProviderService.create(ComponentProviderService.java:62)
>    at org.glassfish.tyrus.core.TyrusServerEndpointConfigurator.<init>(TyrusServerEndpointConfigurator.java:40)
>    at com.oracle.svm.core.reflect.ReflectionAccessorHolder.TyrusServerEndpointConfigurator_constructor_a96715562e86824a6870e2b7a8a7fec2b0cc497d(generated:0)
> ```
> 
> It seems the websocket library is uncompatible any suggestion?

Btw is there any reason to use native image here? Shouldn't Graal's JIT compiler perform better in such a scenario?

**Comment by hamedsharifi** [comment]
native image offer less startup time and much less memory consme I think
